main.tf

```diff
@@ -0,0 +1,213 @@
+#   }
+#   tags = {
+#     Name = "${var.public_subnets}-public*"
+#   }
+# }
+
+# Terraform module which creates S3 Bucket resources for Load Balancer Access Logs on AWS.
+
+module "s3-bucket" {
+  source = "github.com/ministryofjustice/modernisation-platform-terraform-s3-bucket?ref=v6.0.3"
+
+  providers = {
+    aws.bucket-replication = aws.bucket-replication
+  }
+  bucket_prefix       = "${var.application_name}-lb-access-logs"
+  bucket_policy       = [data.aws_iam_policy_document.bucket_policy.json]
+  replication_enabled = false
+  versioning_enabled  = true
+  lifecycle_rule = [
+    {
+      id      = "main"
+      enabled = "Enabled"
+      prefix  = ""
+
+      tags = {
+        rule      = "log"
+        autoclean = "true"
+      }
+
+      transition = [
+        {
+          days          = 90
+          storage_class = "STANDARD_IA"
+          }, {
+          days          = 365
+          storage_class = "GLACIER"
+        }
+      ]
+
+      expiration = {
+        days = 730
+      }
+
+      noncurrent_version_transition = [
+        {
+          days          = 90
+          storage_class = "STANDARD_IA"
+          }, {
+          days          = 365
+          storage_class = "GLACIER"
+        }
+      ]
+
+      noncurrent_version_expiration = {
+        days = 730
+      }
+    }
+  ]
+
+  tags = var.tags
+}
+
+data "aws_iam_policy_document" "bucket_policy" {
+  statement {
+    effect = "Allow"
+    actions = [
+      "s3:PutObject"
+    ]
+    resources = ["${module.s3-bucket.bucket.arn}/${var.application_name}/AWSLogs/${var.account_number}/*"]
+    principals {
+      type        = "AWS"
+      identifiers = [data.aws_elb_service_account.default.arn]
+    }
+  }
+  statement {
+    sid = "AWSLogDeliveryWrite"
+
+    principals {
+      type        = "Service"
+      identifiers = ["delivery.logs.amazonaws.com"]
+    }
+
+    actions = [
+      "s3:PutObject"
+    ]
+
+    resources = ["${module.s3-bucket.bucket.arn}/${var.application_name}/AWSLogs/${var.account_number}/*"]
+
+    condition {
+      test     = "StringEquals"
+      variable = "s3:x-amz-acl"
+
+      values = [
+        "bucket-owner-full-control"
+      ]
+    }
+  }
+
+  statement {
+    sid = "AWSLogDeliveryAclCheck"
+
+    principals {
+      type        = "Service"
+      identifiers = ["delivery.logs.amazonaws.com"]
+    }
+
+    actions = [
+      "s3:GetBucketAcl"
+    ]
+
+    resources = [
+      "${module.s3-bucket.bucket.arn}"
+    ]
+  }
+}
+
+data "aws_elb_service_account" "default" {}
+
+# https://www.terraform.io/docs/providers/aws/d/region.html
+# Get the region of the callee
+data "aws_region" "current" {}
+
+resource "aws_lb" "loadbalancer" {
+  name               = "${var.application_name}-lb"
+  internal           = false
+  load_balancer_type = "application"
+  security_groups    = [aws_security_group.lb.id]
+  subnets            = [var.public_subnets[0], var.public_subnets[1], var.public_subnets[2]]
```

Comment:
Can this just be a list supplied when the module is called?

Reply:
🤔 how do you mean?